### PR TITLE
feat(CNAD): add a new data source cnad advanced available protected o…

### PR DESCRIPTION
**Pull request by deer-hang**
…bjects

**What this PR does / why we need it**:
add a new data source cnad advanced available protected objects
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

This PR depends on #3095. Make sure #3095 merged first.

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cnad' TESTARGS='-run TestAccDatasourceAvailableProtectedObjects_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/cnad -v -run TestAccDatasourceAvailableProtectedObjects_basic -timeout 360m -parallel 4 
=== RUN   TestAccDatasourceAvailableProtectedObjects_basic 
=== PAUSE TestAccDatasourceAvailableProtectedObjects_basic
=== CONT  TestAccDatasourceAvailableProtectedObjects_basic
--- PASS: TestAccDatasourceAvailableProtectedObjects_basic (16.12s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cnad      16.175s


```

### CODE REVIEW

File: docs/data-sources/cnad_advanced_available_protected_objects.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+subcategory: "Cloud Native Anti-DDoS advanced"
+---
+
+# huaweicloud_cnad_advanced_available_protected_objects
```
Comment (niuzhenguo) [inline docs/data-sources/cnad_advanced_available_protected_objects.md]: This name seems too long...

File: docs/data-sources/cnad_advanced_available_protected_objects.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+---
+subcategory: "Cloud Native Anti-DDoS advanced"
```
Comment (niuzhenguo) [inline docs/data-sources/cnad_advanced_available_protected_objects.md]: s/advanced/Advanced

Comment (deer-hang) [reply docs/data-sources/cnad_advanced_available_protected_objects.md]: OK

File: docs/data-sources/cnad_advanced_available_protected_objects.md (hunk 1 above)
Comment (deer-hang) [reply docs/data-sources/cnad_advanced_available_protected_objects.md]: OK, Iwill change it to huaweicloud_cnad_advanced_available_objects

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cnad' TESTARGS='-run TestAccDatasourceAvailableObjects_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/cnad -v -run TestAccDatasourceAvailableObjects_basic -timeout 360m -parallel 4 
=== RUN   TestAccDatasourceAvailableObjects_basic 
=== PAUSE TestAccDatasourceAvailableObjects_basic
=== CONT  TestAccDatasourceAvailableObjects_basic
--- PASS: TestAccDatasourceAvailableObjects_basic (14.74s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cnad      14.790s
```

File: docs/data-sources/cnad_advanced_available_objects.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+subcategory: "Cloud Native Anti-DDoS Advanced"
+---
+
+# huaweicloud_cnad_advanced_available_objects
+
+Use this data source to get the list of CNAD advanced available objects.
```
Comment (ShiChangkuo) [inline docs/data-sources/cnad_advanced_available_objects.md]:
```suggestion
Use this data source to get the list of CNAD advanced available protected objects.
```

File: docs/data-sources/cnad_advanced_available_objects.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+data "huaweicloud_cnad_advanced_available_objects" "test" {
+  instance_id = var.instance_id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `instance_id` - (Required, String) Specifies the instance id.
```
Comment (ShiChangkuo) [inline docs/data-sources/cnad_advanced_available_objects.md]:
```suggestion
* `instance_id` - (Required, String) Specifies the CNAD advanced instance ID.
```

BTW, ID and IP should be capitalized.

File: docs/data-sources/cnad_advanced_available_objects.md (hunk 3 above)
Comment (deer-hang) [reply docs/data-sources/cnad_advanced_available_objects.md]: OK

File: docs/data-sources/cnad_advanced_available_objects.md (hunk 4 above)
Comment (deer-hang) [reply docs/data-sources/cnad_advanced_available_objects.md]: OK

File: docs/data-sources/cnad_advanced_available_objects.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `instance_id` - (Required, String) Specifies the CNAD advanced instance ID. You can find the instance ID through the
+  [document](https://support.huaweicloud.com/api-aad/ListPackage.html).
+  Or find it through data source `huaweicloud_cnad_advanced_instances`.
```
Comment (ShiChangkuo) [inline docs/data-sources/cnad_advanced_available_objects.md]:
```suggestion
  or find it through data source `huaweicloud_cnad_advanced_instances`.
```

BTW, please use the English version of the API document

File: huaweicloud/services/cnad/data_source_huaweicloud_cnad_advanced_available_objects.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the instance id.`,
+			},
+			"protected_object_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the protected object id.`,
+			},
+			"ip": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/cnad/data_source_huaweicloud_cnad_advanced_available_objects.go]: how about use `ip_address`?

Comment (deer-hang) [reply huaweicloud/services/cnad/data_source_huaweicloud_cnad_advanced_available_objects.go]: OK. I have changed `ip` to `ip_address`

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cnad' TESTARGS='-run TestAccDatasourceAvailableObjects_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/cnad -v -run TestAccDatasourceAvailableObjects_basic -timeout 360m -parallel 4 
=== RUN   TestAccDatasourceAvailableObjects_basic 
=== PAUSE TestAccDatasourceAvailableObjects_basic
=== CONT  TestAccDatasourceAvailableObjects_basic
--- PASS: TestAccDatasourceAvailableObjects_basic (17.27s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cnad      17.326s
```

**Pull request merged**